### PR TITLE
feat: add federated catalog samples for an embedded and a standalone federated catalog

### PR DESCRIPTION
**Pull request by farhin23**
## What this PR changes/adds

This PR adds a new module `federated-catalog`, which includes the following samples,
* `fc-00-basic`:  includes preparations for the later Federated Catalog samples
* `fc-01-embedded`: demonstration of a federated catalog, embedded in a connector.
* `fc-02-standalone`: demonstrates the implementation of a standalone catalog.

## Why it does that

To illustrate how we can implement different types of Federated Catalogs, e.g embedded and standalone.

## Further notes

Projects and libraries have been added in [settings.gradle.kts](https://github.com/eclipse-edc/Samples/blob/main/settings.gradle.kts), and [gradle/libs.versions.toml](https://github.com/eclipse-edc/Samples/blob/main/gradle/libs.versions.toml).

## Linked Issue(s)
Contributes to #296

_Please be sure to take a look at the [contributing guidelines](https://github.com/eclipse-edc/.github/blob/main/CONTRIBUTING.md#submit-a-pull-request) and our [etiquette for pull requests](https://github.com/eclipse-edc/.github/blob/main/contributing/pr_etiquette.md)._

### CODE REVIEW

**Review by ndr-brt — CHANGES_REQUESTED** [review]
please provide automated tests on the `system-tests` module that ensure the samples are working correctly, as done for the other samples

**Review by ndr-brt — CHANGES_REQUESTED** [review]

File: federated-catalog/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+---
+
+### Different Implementations of Node Resolver
+
+
+In the following samples you will learn how to implement a TargetNodeDirectory depending on different use cases.
+
+### [FC sample 03](): Resolve Target Catalog Nodes from static participant file
+This sample demonstrates a Catalog Node resolver, that implements TargetNodeDirectory. It resolves the Target Catalog
+Nodes from a static participant file containing the DSP endpoints of the participants.
```
[the remark this answers is not in the document]
Comment (farhin23) [reply federated-catalog/README.md]: This block has been removed and will be introduced in later PRs.

File: federated-catalog/README.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+
+The following samples shows how to
+* implement, build and run different versions of FC e.g.
+    * standalone,
+    * embedded.
+
+[//]: # (* implement TargetNodeDirectory and resolve Target Nodes,)
+
+[//]: # (    * from a static file containing all participants' DSP endpoints,)
```
Comment (ndr-brt) [inline federated-catalog/README.md:21]: I'm not an huge fan of comments, they tend to be ignored and stale and become outdated.
Please remove them and just create an issue in this repository reporting what needs to be done.

**Review by ndr-brt — CHANGES_REQUESTED** [review]

**Comment by ndr-brt** [comment]
> The pipeline tests are experiencing unpredictable failures. In the latest commit, one of the tests in `Policy01Basic` fails despite no changes being made in this area.
> 
> **Probable cause:** Since this is a time-dependent test, a likely cause could be the repetitive tasks in FederatedCatalog runtimes that continue running even after the associated runtimes are shut down. Although the shutdown methods for RuntimeExtension terminate the runtimes, they do not stop the `ScheduledExecutorService` responsible for recursive crawling within FederatedCatalog, which remains active and continues to take up resources. This might be causing delays in the execution of the Policy01Basic test, ultimately leading to its failure.

this is definitely an issue that needs to be fixed in the FederatedCatalog repo, would you mind to open an issue? (end eventually provide a PR to fix it?)

> **Solution:** Proper shutdown method in `FederatedCatalogCacheExtension` to trigger the shutdown of `ScheduledExecutorService` should solve this issue. This will ensure that all the scheduled tasks are terminated properly at the time of runtime shutdown. Regardless of the current pipeline failure, the scheduled tasks in `FederatedCatalog` requires a proper shutdown service.

that's correct, but we need a workaround right here to make it pass, like increasing the timeout or increasing the "period seconds" setting in the FC runtime configuration

**Review by ndr-brt — APPROVED** [review]
anyway, it just passed, so LGTM and please open an issue in the FCC repository about the thread shutdown. I guess tests will remain flake until the fix will be rolled out in a new version. Given that this is the Samples repos I guess we can bear with that.

**Comment by farhin23** [comment]
> this is definitely an issue that needs to be fixed in the FederatedCatalog repo, would you mind to open an issue? (end eventually provide a PR to fix it?)
> 
I will open an issue and take care of the fix.

> that's correct, but we need a workaround right here to make it pass, like increasing the timeout or increasing the "period seconds" setting in the FC runtime configuration

I will leave it for now, as the current commit just passed. And will make necessary adjustments in future commits.

**Pull request merged**